.github/workflows/test.yaml

```diff
@@ -257,12 +257,29 @@ jobs:
           fi
 
 
+  # We have this job so that the PR can be blocked on a single job.
+  # In this way, each time a job is modified,
+  # we don't have to go to GitHub UI and reconfigure branch protection.
+  # See GitHub support answer for this hack.
+  # https://gist.github.com/altendky/2e3483a1f7e1ba21cc97de75db9b7d1c
   all-successful:
-    # We have this job so that the PR can be blocked on a single job.
-    # See GitHub support answer for this hack.
-    # https://gist.github.com/altendky/2e3483a1f7e1ba21cc97de75db9b7d1c
+    # Is very important to force running this always, as otherwise it will be
+    #  skipped by default.
+    if: always()
     runs-on: ubuntu-latest
-    needs: [testing, static-checks, release-publish]
+    # Here should be the list of all the other jobs defined in this file.
+    needs:
+      - testing
+      - static-checks
+      - release-publish
     steps:
-    - name: note that all tests succeeded
-      run: echo "🎉"
+      - name: Require all successes
+        shell: python
+        env:
+          RESULTS: ${{ toJSON(needs.*.result) }}
+        run: |
+          import json
+          import os
+          import sys
+          results = json.loads(os.environ["RESULTS"])
+          sys.exit(0 if all(result == "success" for result in results) else 1)
```

Comment:
As I see this show up in more places, maybe it should be made into an action?  Or the whole job into a reusable workflow.  In this case the reusable workflows are used to get around the 256 job per matrix limit, but it is still an example with the `needs:` in use.

https://github.com/Chia-Network/chia-blockchain/blob/9a4316f96eeb92a06d5ceaf94387d349c5ea9537/.github/workflows/test.yml#L45-L55
```yaml
  macos:
    uses: ./.github/workflows/test-single.yml
    needs: configure
    with:
      emoji: 🍎
      matrix: macos
      name: macOS
      file_name: macos
      concurrency_name: macos
      configuration: ${{ needs.configure.outputs.configuration }}
      runs-on: macos-latest
```

Reply:
I have merged this as it should be better than what we have now.

I think that a shared action should be good enough.

--------

I am not sure how workflow interdependencies work.

Reply:
Thanks... I think that is nice, but for Twisted is a bit complicated.

I mean the current code, should be easier to read and understand by average Twisted dev that is not an expert into GiHub Actions.

Reply:
> As I see this show up in more places, maybe it should be made into an action?

Yes: https://github.com/re-actors/alls-green